### DISCUSSION TITLE
Allow to run out of the Form context

### DISCUSSION THREAD

**Discussion by artola** [edmundhung/conform#723] [Ideas]
The hook throws if can not find the Form context. It would be convenient to allow a config/option to avoid throwing. In such a way we can add the logic to a base component, independently if there is a form or not around it.

https://github.com/edmundhung/conform/blob/6b98c077d757edd4846321678dfb6de283c177b1/packages/conform-react/context.tsx#L139

**Comment by edmundhung** [comment]
Can you help me understand better about your usecase? Removing this check means everything consuming the metadata from the form context (e.g. `useField`, `useFormMetadata`) might be `undefined` which would be quite messy to work with.

**Reply by artola** [reply]
We use to create components using "react-aria" approach, using the context to pass props to the composed components.
In general, working with the React context can follow one of these approaches:

- if you can not find the context, you throw (e.g., react-router, a route/link throws if it can not find the context).
- if you can not find the context, you ignore it, it means that it is not in such composition.

ATM my colleague did create a custom hook with a try/catch to workaround this thing. Otherwise, we should duplicate/patch the code of the hook to not throw.

FYI we are build a `<Form />` that supports a schema validation, and it passes down to the `<Field />`, using the context, the validation result. Why? We avoid the wiring and spreading of props, making components almost a black box, the consumer does not even know which validation library is using, lowering the entry level and making a lot cleaner the code (less lines, less implementation details).
